master/internal/rm/kubernetesrm/informer_intg_test.go

```diff
@@ -213,12 +180,165 @@ func TestNodeInformer(t *testing.T) {
 					Object: node,
 				}
 			}
+
+			wg.Wait()
+			assert.Equal(t, tt.output, currNodes)
+		})
+	}
+}
+
+func TestEventListener(t *testing.T) {
+	cases := []struct {
+		name       string
+		expected   error
+		eventNames []string
+	}{
+		{"zero events", nil, []string{}},
+		{"listener success", nil, []string{"A"}},
+		{
+			"listener success & event ordering success",
+			nil,
+			[]string{"A", "B", "C", "D", "E"},
+		},
+		{
+			"listener success & event ordering failure",
+			nil,
+			[]string{"A", "B", "C", "D", "E"},
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			var wg sync.WaitGroup
+			wg.Add(len(tt.eventNames))
+
+			ctx := context.TODO()
+			eventChan := make(chan watch.Event)
+			ordering := make([]string, 0)
+
+			mockOptsList, mockOptsWatch := initializeMockOpts("event")
+			mockEventInterface := &mocks.EventInterface{}
+
+			mockEventInterface.On("List", ctx, mockOptsList).Return(
+				&k8sV1.EventList{
+					ListMeta: metaV1.ListMeta{
+						ResourceVersion: "1",
+					},
+				},
+				nil)
+			mockEventInterface.On("Watch", ctx, mockOptsWatch).Return(&mockWatcher{c: eventChan}, nil)
+
+			mockEventHandler := func(event *k8sV1.Event) {
+				t.Logf("received event %v", event)
+				ordering = append(ordering, event.Name)
+				wg.Done()
+			}
+
+			i, err := newEventListener(
+				ctx,
+				mockEventInterface,
+				namespace,
+				mockEventHandler)
+			if err != nil {
+				assert.Nil(t, i)
+				assert.Error(t, tt.expected, err)
+				return
+			}
+			assert.NotNil(t, i)
+			assert.Equal(t, tt.expected, err)
+
+			go i.run()
```

Comment:
can do it as a followup, but i would make `run` take a context so it is cancelable (since we aren't doing that, if tests hang we are going to have a lot of unrelated goroutines in the stack)

Reply:
i'll add that as a note in the next iteration of this ticket -- and yes, I have noticed that if the tests hang its a pain to actually find what failed.